exec/java-exec/src/main/java/org/apache/drill/exec/store/ischema/Records.java

```diff
@@ -503,12 +568,97 @@ public Column(String catalog, String schemaName, String tableName, RelDataTypeFi
           this.COLUMN_SIZE = null;
         break;
       }
+      this.COLUMN_FORMAT = null;
+      this.NUM_NULLS = null;
+      this.MIN_VAL = null;
+      this.MAX_VAL = null;
+      this.NDV = null;
+      this.EST_NUM_NON_NULLS = null;
+    }
 
+    public Column(String catalog, String schemaName, String tableName, String columnName,
```

Comment:
Looks like this constructor contains well-refactored code with some logic similar to the logic in the constructor above. Is it possible to combine these constructors?

Reply:
Both constructors cannot be combined since they determine `Record.Column` properties based on different sources. One uses `RelDataTypeField`, another `ColumnMetadata`, `ColumnStatistics` and additional properties, mostly relying on `TypeProtos.MajorType`. Logic go obtaining `Record.Column` is different, so I suggest we leave both constructors.

Personally, I would prefer to remove some of the properties from `Record.Column`, like `NUMERIC_PRECISION_RADIX` etc. since I am not sure how much value they bring but I had to leave them for backward compatibility.

Reply:
Ok, thanks for the explanation.